google-http-client/src/main/java/com/google/api/client/http/HttpResponse.java

```diff
@@ -432,8 +432,10 @@ public void ignore() throws IOException {
    * @since 1.4
    */
   public void disconnect() throws IOException {
-    ignore();
+    // Close the connection before trying to close the InputStream content. If you are trying to
+    // disconnect, we shouldn't need to try to read any further content.
     response.disconnect();
+    ignore();
```

Comment:
Why do you need to call this at all? I'd think disconnecting alone is enough.

Reply:
I'm not sure of the original intention for sure, but it looks like it's supposed to close the response's `InputStream` so you can't read from it anymore - a way of letting google-http-client cleaning up the connections and their resources. This class was designed before `AutoClosable` was available (and possibly even before `Closable` was available).

Reply:
The more I look at this, the more off this seems. I think what we should do is:

```
      InputStream lowLevelResponseContent = this.response.getContent();
      lowLevelResponseContent.close();
``
`ignore()` actually reads and decodes the bytes remaining on the stream. That could be quite a long-running operation. The point of disconnecting is to throw those away and not waste time reading them.

If you see otherwise, let's discuss.

Reply:
That seems reasonable, I'll try it out.

Reply:
I modified `ignore()` to directly close the `LowLevelHttpResponse`'s `InputStream` rather than doing the extra reading logic in `getContent()`.

I still do need to swap the `disconnect()` and `ignore()` lines as the `ApacheHttpResponse` will still try to read the content from its input stream unless it's disconnected first.